module/VuFind/src/VuFind/Controller/AbstractBase.php

```diff
@@ -179,6 +180,35 @@ protected function createViewModel($params = null)
         return new ViewModel($params);
     }
 
+    /**
+     * Execute the request
+     *
+     * @param MvcEvent $e MVC event object
+     *
+     * @return mixed
+     *
+     * @throws Exception\DomainException
+     */
+    public function onDispatch(MvcEvent $e)
```

Comment:
Not every controller extends AbstractBase in VuFind (currently, most do, but that may change with future refactoring when time is found to move away from serviceLocator-dependent helper methods in favor of more disciplined dependency injection). I wonder if it would be more future-proof to set up this functionality through the VuFind\Bootstrapper rather than relying on specific inheritance patterns.

(This is just a suggestion; it will probably work as-is for now... but I just want to be sure we're applying this as thoroughly and completely as possible).

Reply:
No problem, I refactored into Bootstrapper. I think, it should be ideally implemented as PSR15 middleware. But it would need much more infrastructure and refactoring definitely out of the scope of this PR. But may be worth thinking about it for the future.

Reply:
Yes, that might be a good project for VuFind 8.0; feel free to open a JIRA "architecture" ticket if you want a place to discuss the idea further.